staging/src/k8s.io/kubectl/pkg/cmd/config/set_test.go

```diff
@@ -54,6 +57,1183 @@ func TestSetConfigCurrentContext(t *testing.T) {
 	test.run(t)
 }
 
+func TestSetConfigAuthProviderName(t *testing.T) {
```

Comment:
Can we please make these table driven tests to cut down on duplication?

Reply:
Hopefully what I did is what you wanted, but it didn't really seem to save us many lines. Apologies if I've misunderstood.